File: php/features.php
<?php
declare(strict_types=1);

// AirQuality SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AirQualityFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AirQualityBaseFeature();
            case "test":
                return new AirQualityTestFeature();
            default:
                return new AirQualityBaseFeature();
        }
    }
}

File: py/airquality_sdk/features.py
# AirQuality SDK feature factory

from airquality_sdk.feature.base_feature import AirQualityBaseFeature
from airquality_sdk.feature.test_feature import AirQualityTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AirQualityBaseFeature(),
        "test": lambda: AirQualityTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
